util/dvsim/Deploy.py

```diff
@@ -524,9 +542,10 @@ def __init__(self, build_mode, sim_cfg):
         CompileSim.items.append(self)
 
     def dispatch_cmd(self):
-        # Delete previous cov_db_dir if it exists before dispatching new build.
+        # Delete old coverage database directories before building again. We
+        # need to do this becuase build directory is not 'renewed'.
         if os.path.exists(self.cov_db_dir):
-            os.system("rm -rf " + self.cov_db_dir)
+            shutil.rmtree(self.cov_db_dir)
```

Comment:
Maybe it would make sense to move this logic into a helper function, since it gets used in a couple of places. Also, the usual way to do this is to try to delete, ignoring any exception, then check the directory no longer exists.

Reply:
Well, it will run marginally faster (avoiding a system call) and the "try to do it and handle any problems" approach seems to be more idiomatic Python.